pkg/controllers/kube-apiserver.go

```diff
@@ -111,6 +112,54 @@ func (s *KubeAPIServer) configure(cfg *config.Config) error {
 	s.servingCAPath = cryptomaterial.ServiceAccountTokenCABundlePath(certsDir)
 	s.advertiseAddress = cfg.ApiServer.AdvertiseAddress
 
+	namedCerts := []configv1.NamedCertificate{
+		{
+			CertInfo: configv1.CertInfo{
+				CertFile: cryptomaterial.ServingCertPath(cryptomaterial.KubeAPIServerExternalServingCertDir(certsDir)),
+				KeyFile:  cryptomaterial.ServingKeyPath(cryptomaterial.KubeAPIServerExternalServingCertDir(certsDir)),
+			},
+		},
+		{
+			CertInfo: configv1.CertInfo{
+				CertFile: cryptomaterial.ServingCertPath(cryptomaterial.KubeAPIServerLocalhostServingCertDir(certsDir)),
+				KeyFile:  cryptomaterial.ServingKeyPath(cryptomaterial.KubeAPIServerLocalhostServingCertDir(certsDir)),
+			},
+		},
+		{
+			CertInfo: configv1.CertInfo{
+				CertFile: servingCert,
+				KeyFile:  servingKey,
+			},
+		},
+	}
+	if len(cfg.ApiServer.NamedCertificates) > 0 {
+		for _, namedCertsCfg := range cfg.ApiServer.NamedCertificates {
+			//Validate the cert is non-destructive
+			certAllowed, err := util.IsCertAllowed(cfg.ApiServer.AdvertiseAddress, cfg.Network.ClusterNetwork, cfg.Network.ServiceNetwork, namedCertsCfg.CertPath, namedCertsCfg.Names)
+			if err != nil {
+				klog.Warningf("Failed to read NamedCertificate - ignoring: %v", err)
+				continue
+			}
+
+			if !certAllowed {
+				klog.Warningf("Certificate %v not allowed and ignored", namedCertsCfg)
```

Comment:
Comment style similar to above
```suggestion
				klog.Warningf("Certificate %v is not allowed - ignoring", namedCertsCfg)
```